src/modules/external/aha/components/AltAhaScore.tsx

```diff
@@ -0,0 +1,103 @@
+import { Stack, Typography } from '@mui/material';
+import { useCallback, useContext, useState } from 'react';
+import LabelWithContent from '@/components/elements/LabelWithContent';
+import LoadingButton from '@/components/elements/LoadingButton';
+import AssessmentContext from '@/modules/assessments/components/AssessmentContext';
+import ApolloErrorAlert from '@/modules/errors/components/ApolloErrorAlert';
+import ErrorAlert from '@/modules/errors/components/ErrorAlert';
+import { emptyErrorState, ErrorState, hasErrors } from '@/modules/errors/util';
+import { FormDefinitionHandlers } from '@/modules/form/hooks/useFormDefinitionHandlers';
+import { DynamicInputCommonProps } from '@/modules/form/types';
+import {
+  CalculateAltAhaScoreMutation,
+  useCalculateAltAhaScoreMutation,
+} from '@/types/gqlTypes';
+
+interface AltAhaScoreProps extends DynamicInputCommonProps {
+  value?: number | null;
+  onChange?: (value: number | null) => void;
+  handlers?: FormDefinitionHandlers;
+}
+
+// Shows a button to calculate the score based on current form values, and displays the result.
+const AltAhaScore = ({
+  value,
+  onChange,
+  label,
+  disabled = false,
+  handlers,
+}: AltAhaScoreProps) => {
+  const [errorState, setErrorState] = useState<ErrorState>(emptyErrorState);
+
+  const [calculateAltAhaScore, { loading }] = useCalculateAltAhaScoreMutation({
+    onCompleted: (data: CalculateAltAhaScoreMutation) => {
+      const errors = data.calculateAltAhaScore?.errors || [];
+      if (errors.length > 0) {
+        setErrorState({ ...emptyErrorState, errors });
+        return;
+      }
+
+      setErrorState(emptyErrorState);
+
+      if (onChange && data.calculateAltAhaScore) {
+        onChange(data.calculateAltAhaScore.score || null);
+      }
+    },
+    onError: (apolloError: any) => {
+      setErrorState({ ...emptyErrorState, apolloError });
+    },
+  });
+
+  const { formDefinitionIdentifier, enrollmentId } =
+    useContext(AssessmentContext) || {};
+
+  const handleFetch = useCallback(() => {
+    if (!enrollmentId || !formDefinitionIdentifier) {
+      // In the preview environment, mock the response instead of calling the mutation
+      onChange?.(10);
+    } else {
```

Comment:
This worries me because if there is a render issue with the context, or this is rendered somewhere in the app without a context, the user will see "10" as the score (and perhaps even get that value stored for the client?). Suggest raising instead:

```suggestion
    if (!enrollmentId || !formDefinitionIdentifier) {
      throw new Error(
        'unable to calculate alt-aha without assessment context values'
      );
    } else {
```

Reply:
Or maybe the button should be disabled? (to avoid sentry noise, since users are likely to click the button in preview)

Reply:
Sure that seems OK. I'm not too concerned about previews, I guess, at some point we should dedicate some time to making it more robust. (For example, provide an easy way for the form component to know that its being rendered in preview)